core/domain/stats_services.py

```diff
@@ -31,6 +32,68 @@
 VERSION_ALL = 'all'
 
 
+def _migrate_issue_schema(exp_issue_dict):
+    """Holds the responsibility of performing a step-by-step sequential update
+    of an exploration issue structure based on the schema version of the input.
+    If the current issue schema version changes
+    (stats_models.CURRENT_ISSUE_SCHEMA_VERSION), a new conversion function must
+    be added and some code appended to this function to account for that new
+    version.
+
+    Args:
+        exp_issue_dict: dict. Dict representing the exploration issue.
+
+    Raises:
+        Exception. The issue_schema_version is invalid.
+    """
+    issue_schema_version = exp_issue_dict['schema_version']
+    if issue_schema_version is None or issue_schema_version < 1:
+        issue_schema_version = 0
+
+    if not (0 <= issue_schema_version
+            <= stats_models.CURRENT_ISSUE_SCHEMA_VERSION):
+        raise Exception(
+            'Sorry, we can only process v1-v%d and unversioned issue schemas at'
+            'present.' %
+            stats_models.CURRENT_ISSUE_SCHEMA_VERSION)
+
+    while issue_schema_version < stats_models.CURRENT_ISSUE_SCHEMA_VERSION:
+        stats_domain.ExplorationIssue.update_exp_issue_from_model(
+            exp_issue_dict)
+        issue_schema_version += 1
+
+
+def _migrate_action_schema(learner_action_dict):
```

Comment:
ditto: naming (there should be a notion of "latest")

Reply:
Done.